### ISSUE TITLE
Color Contrast discrepancy

### ISSUE THREAD

**Issue by kevinmpowell** [chanzuckerberg/axe-storybook-testing#36]
I'm getting a color contrast failure on a test when running this package that I don't receive in the browser. The test environment is "seeing" different hex codes than what are set in code. To debug I tried changing the failing foreground color to css 'hotpink': `#ff69b4`. 

Here are the values I get in the error message across all three test browsers:

| Actual Hex Code | Chromium | Firefox | Webkit |
| --- | --- | --- | ---|
| `#ff69b4` | `#fedd9c` | `#ff9aaa` | `#ff71b2` |

None of them match the actual hex code. I have no idea what's happening here. Any thoughts?

**Comment by ahuth** [comment]
Weird! Thanks for the issue report.

Couple questions:
- Is there any transparency or opacity on any elements in the component?
- Are there any animations or transitions (especially on opacity)?
- Are you using storybook's a11y addon? If so, do you see the failure if you change it to be the first addon (so that it runs immediately when the story renders)?

**Comment by kevinmpowell** [comment]
> * Is there any transparency or opacity on any elements in the component?
No. These are solid hex codes in CSS.

> * Are there any animations or transitions (especially on opacity)?
No, I've disabled all animations.

> * Are you using storybook's a11y addon? If so, do you see the failure if you change it to be the first addon (so that it runs immediately when the story renders)?
I'm not exactly sure how to do this. I put it first in the list of addons in `.storybook/main.js`. Is that all it takes?
<img width="332" alt="image" src="https://user-images.githubusercontent.com/979463/134054972-8357dd66-c296-4c0f-9264-73e55b03bbfa.png">

**Comment by ahuth** [comment]
> I put it first in the list of addons in .storybook/main.js. Is that all it takes?

Yes, I believe that's all it takes.

Also, can you:
- Post a screen shot of the component rendered in storybook (if you're able. I'm assuming it's not secret or anything)
- Let me know which version of axe-core you're using

**Comment by kevinmpowell** [comment]
<img width="533" alt="image" src="https://user-images.githubusercontent.com/979463/134057602-31f1e260-baaf-415e-85f2-0eea9cf3291f.png">
In the screenshot I reverted the text from `hotpink` back to `#002B3F`.


**Version**
`axe-core@4.3.3`

**Full error message**
```
1. color-contrast (Elements must have sufficient color contrast)
       
          For more info, visit https://dequeuniversity.com/rules/axe/4.3/color-contrast?application=axeAPI.
       
          Check these nodes:
       
          - html: <p class="MuiTypography-root MuiTypography-body1 css-9l3uo3">Use the <strong>Controls</strong> tab in addons panel to modify the story</p>
            summary: Fix any of the following:
                       Element has insufficient color contrast of 1.4 (foreground color: #d2c48a, background color: #fee59a, font size: 12.0pt (16px), font weight: normal). Expected contrast ratio of 4.5:1
       
          - html: <strong>Controls</strong>
            summary: Fix any of the following:
                       Element has insufficient color contrast of 1.4 (foreground color: #d2c48a, background color: #fee59a, font size: 12.0pt (16px), font weight: bold). Expected contrast ratio of 4.5:1
```

**Comment by ahuth** [comment]
Thanks for the info. It's weird that the foreground color is showing up as `#d2c48a`.

Couple more things to try
- If you run the tests again, does the `#d2c48a` change or stay the same? If it changes, that usually means there's some sort of transition.
- If you run non-headlessly with the `--headless=false` CLI flag, does anything change?
- What version of axe-storybook-testing are you on? v4 upgraded the version of Playwright that's used (Not sure if that would effect this or not)
- Can you share what styles are applying to the element? (Only if the code is open source)

**Comment by kevinmpowell** [comment]
> * If you run the tests again, does the `#d2c48a` change or stay the same? If it changes, that usually means there's some sort of transition.

I switched to a different branch and then back. Oddly the foreground changed to `#eed994`. I've now re-run the tests three times and consistently get `#eed994`.

> * If you run non-headlessly with the `--headless=false` CLI flag, does anything change?

`--headless=false` still results in: `#eed994`

But headless false reveals a transition I wasn't aware of. The element is fading in.
<img width="514" alt="image" src="https://user-images.githubusercontent.com/979463/134067508-d4890ece-0268-4f5c-a2b6-005ef0633d8f.png">

> * What version of axe-storybook-testing are you on? v4 upgraded the version of Playwright that's used (Not sure if that would effect this or not)

`@chanzuckerberg/axe-storybook-testing@4.0.1`

> * Can you share what styles are applying to the element? (Only if the code is open source)

I'll dig into this if I can't fix the transition issue that seems to be causing the problem

**Comment by kevinmpowell** [comment]
@ahuth thanks so much for helping me troubleshoot. Disabling the transition solved the failing test.

BTW, I'm loving this storybook/a11y/CI solution. Wish I'd found it sooner.

**Comment by ahuth** [comment]
No problem!

These tests run with [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) forced to on. So one thing you can do is turn off any transitions when the `prefers-reduced-motion` media query matches.

Glad you're getting use out of this project!
